staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight_test.go

```diff
@@ -319,3 +319,43 @@ func TestMaxInFlightSkipsMasters(t *testing.T) {
 
 	responses.Wait()
 }
+
+func TestDisabledMaxInFlightWatermarkMainteneace(t *testing.T) {
+	allowedNonMutatingInflightRequestsNo := 3
+
+	// Calls is used to wait until all server calls are received. We are sending
+	// AllowedNonMutatingInflightRequestsNo of 'long' not-accounted requests and the same number of
+	// 'short' accounted ones.
+	calls := &sync.WaitGroup{}
+	calls.Add(allowedNonMutatingInflightRequestsNo * 2)
+
+	// Responses is used to wait until all responses are
+	// received. This prevents some async requests getting EOF
+	// errors from prematurely closing the server
+	responses := &sync.WaitGroup{}
+	responses.Add(allowedNonMutatingInflightRequestsNo * 2)
+
+	// Block is used to keep requests in flight for as long as we need to. All requests will
+	// be unblocked at the same time.
+	block := &sync.WaitGroup{}
+	block.Add(1)
+
+	waitForCalls := true
+	waitForCallsMutex := sync.Mutex{}
+
+	server := createMaxInflightServer(calls, block, &waitForCalls, &waitForCallsMutex, 1, allowedNonMutatingInflightRequestsNo, false)
+	defer server.Close()
+
+	// when we disable watermarks that mean that we do not write to inFlightWatermark variable
```

Comment:
See the comments I made about the TestDisabledApfWatermarkMainteneace test.

Reply:
reworked in 9d2e3fb8557c66cde87223933cdedb599fd33736